### ISSUE TITLE
requests from the same host do not pass

### ISSUE THREAD

**Issue by Miskler** [benoitc/gunicorn#3041]
I have two applications on my server: 
1. FastAPI application running through gunicorn installed via pip. I run it like this:
```py
gunicorn main:app -b 5.44.41.136:8000 --access-logfile access.log --error-logfile error.log -c gunicorn_config.py --worker-class uvicorn.workers.UvicornWorker
```
`gunicorn_config.py`:
```py
bind = "0.0.0.0:443"
certfile = "/some_way/43093.zetalink.ru/cert.pem"
keyfile = "/some_way/43093.zetalink.ru/privkey.pem"
workers = 1
```
Requests that come from other computers are working. But if I'm on the same VPS as my application, I'll launch another one as follows:
```py
import requests

#I also tried using the domain in the request 43093.zetalink.ru and ip in all variants (both http and https connection)
o = requests.get(url="http://5.44.41.136:8000/docs")
```
`sudo python3 test.py`
I get the following error:
```
Traceback (most recent call last):
  File "/home/steam/pytorrent/telegram_bot/test.py", line 3, in <module>
    o = requests.get(url="http://5.44.41.136:8000/docs")
  File "/usr/local/lib/python3.10/dist-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 547, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

Both applications are located on the same VPS server in neighboring folders. Literally an hour ago, the reinstallation of my server ended in order to understand that no other programs interfere with the operation of my server. I have installed the following packages:
```
htop
git
screen
python3.10
pip
python dependencies of my applications, including: pip install gunicorn
``` Os Ubuntu 22.04, all operating system packages have been updated to the latest version.

**Comment by Miskler** [comment]
I'm sure it's gunicorn because here is the log of the server console responses when trying to connect to itself:
![image](https://github.com/benoitc/gunicorn/assets/69055489/1079a6b3-8124-4ac9-bfa1-b7aef769c4b1)

**Comment by Miskler** [comment]
As well as the log of responses when requested to the localhost address
![image](https://github.com/benoitc/gunicorn/assets/69055489/149e5a93-f26b-4792-a2ad-62fb26a17699)

**Comment by Miskler** [comment]
As a result, the error was due to the following reasons:
1. I specified a public IP at startup, which means that the IP addresses in the `localhost` group _(such as `0.0.0.0` and `127.0.0.1`)_ are not listened to.
2. I specified the wrong certificate. `cert.pem` was enough for the browser, but Ubuntu tried to verify the authenticity of the certificate. I replaced `cert.pem` in the settings with `fullchain.pem`.

At this stage, I am glad that I was able to solve the problem, but I am also very disappointed that I spent 2 days on it and had to reinstall the system. :(
